### PR TITLE
(#281) Bump GH-Action images to latest

### LINKED ISSUE #281: Build pipeline does not run correctly
image

Identified so far:

- Build (windows-2019) - windows-2019 is missing .NET 5.0
- Build (macos-10.15) - macos-10.15 does no longer exist
- Build (ubuntu-20.04) - ubuntu-20.04 does no longer exist

### LINKED ISSUE #283: Update Cake.Recipe to latest 3.x version
We should update the version of Cake.Recipe being used to the latest available version of Cake.Recipe. This will allow us to make use of some of the features introduced in a later version there.

### PR DESCRIPTION
**Pull request by nils-a**
and added the required dotnet versions.

fixes #281 
fixes #283

### CODE REVIEW

**Comment by nils-a** [comment]
@AdmiringWorm Everything is working again, apart from mergify which seems to be broken because of https://blog.mergify.com/strict-mode-deprecation/. However, I have no idea how to fix that in a meaningful way.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -43,14 +43,22 @@ jobs:
         uses: actions/cache@v2.1.5
         with:
           path: tools
-          key: ${{ runner.os }}-tools-${{ hashFiles('setup.cake') }}
+          key: ${{ runner.os }}-tools-${{ hashFiles('setup.cake') }}-${{ hashFiles('.config/dotnet-tools.json') }}
```
Comment (AdmiringWorm) [inline .github/workflows/build.yml:46]: The version of cake is the only dotnet-tool used, and it is explicitly set in this workflow, as such there is no need to change the cache key here.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -17,7 +17,7 @@ jobs:
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        os: [windows-2019, ubuntu-18.04, macos-10.15]
+        os: [windows-2022, ubuntu-22.04, macos-12]
```
Comment (AdmiringWorm) [inline .github/workflows/build.yml:20]:
```suggestion
        os: [windows-2022, ubuntu-22.04, macos-13]
```

**Review by AdmiringWorm — CHANGES_REQUESTED** [review]
Thanks for putting this PR up, a few changes but I'll take care of them before merging it in.

**Comment by AdmiringWorm** [comment]
This have been merged in commit https://github.com/cake-contrib/Cake.Codecov/commit/2796c0255f1e1ac788753bc5f587c55b28b67bb0

**Comment by AdmiringWorm** [comment]
@all-contributors please add @nils-a for code

**Pull request closed without merging**